public/basket.html

```diff
@@ -334,26 +334,27 @@ <h4 class="modal-title" id="credit-card">Credit Card</h4>
         });
     }
 
-    function updateCart() {
-        console.log('Updating cart.');
-        $("tr.item").each(function (){
-            var id = $(this).find("input.id").val(),
-                quantity = $(this).find("input.form-control").val();
-            console.log("Item nr " + id  + " has " + quantity);
-            $.ajax({
-                url: "cart/" + id,
-                type: "PUT",
-                data: JSON.stringify({"quantity": quantity}),
-                success: function (data, textStatus, jqXHR) {
-                    console.log('Worked!');
-                    location.reload();
-                },
-                error: function (jqXHR, textStatus, errorThrown) {
-                    console.error('Could not update item: ' + id + ', due to: ' + textStatus + ' | ' + errorThrown);
-                }
-            });
-        });
-    }
+    // function updateCart()
+	// for each item row in cart table call updateToCart (client.js - updateToCart(itemId, quantity, callback))
+	function updateCart() {
+		console.log("Updating Cart");
+		var cartsize = document.getElementById("cart-list").rows.length;
+		console.log("cart-list size: " + cartsize);
+
+		var idx = 0;
+		next = function(){
+			if (idx< cartsize) {
+				var id = document.getElementById("cart-list").rows[idx].cells[2].id;
+				var quantity = document.getElementById("cart-list").rows[idx].cells[2].getElementsByTagName('input')[0].value;
+				idx++;
+				updateToCart(id, quantity, next);
+			}
+			else {
+				location.reload();
+			}
+		}
+		next();
```

Comment:
Not a good idea. It is the entry point to the callback routine that is called recursively.